src/common/baseTelemetryReporter.ts

```diff
@@ -309,9 +326,58 @@ export class BaseTelemetryReporter {
 	 * @param measurements The measurements (numeric values) to send with the event
 	 */
 	public sendRawTelemetryEvent(eventName: string, properties?: RawTelemetryEventProperties, measurements?: TelemetryEventMeasurements): void {
-		if (this.userOptIn && eventName !== "") {
```

Comment:
keeping the OptIn checks in the outer methods would let you reduce the number of parameters required for the internal methods

Reply:
Yeah but then I would need it in the raw and normal log functions. Here I only need it in the internal one. I'm not really worried about the number of params. If it keeps getting large I'll probably accept an `options` bag this like `{sanitize: true, dangerous: false}`